x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CachedBlobContainerIndexInput.java

```diff
@@ -131,6 +172,31 @@ protected void readInternal(final byte[] buffer, final int offset, final int len
         lastSeekPosition = lastReadPosition;
     }
 
+    /**
+     * Prefetches a complete part and writes it in cache. This method is used to prewarm the cache.
+     */
+    public int prefetchPart(final int part) throws IOException {
+        ensureContext(ctx -> ctx == CACHE_WARMING_CONTEXT);
+        if (part >= fileInfo.numberOfParts()) {
+            throw new IllegalArgumentException("Unexpected part number [" + part + "]");
+        }
+        final Tuple<Long, Long> range = computeRange(IntStream.range(0, part).mapToLong(fileInfo::partBytes).sum());
```

Comment:
Can we assert that the returned range is the whole part? I think that'd be useful documentation if nothing else.

Reply:
Sure. I added e01bf710ffa6a00ab730fc3efe45b4204dadd763, let me know what you think.